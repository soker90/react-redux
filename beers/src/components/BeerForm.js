import React, { Component } from 'react'

class BeerForm extends Component {

  handleSubmit (e) {
    e.preventDefault()
    this.props.addBeer(this.refs.name.value, this.refs.graduation.value)
    console.log(this.refs.name.value)
  }

  render () {
    const btnText = this.props.loading ? 'Loading...' : 'Add Beer'

    return (
      <div>
        <form action="#" onSubmit={(e) => this.handleSubmit(e)}>
          <input type="text" ref="name" name="beerName" required="required" />
          <input type="text" ref="graduation" name="beerGraduation" required="required" />
          <input type="submit" value={btnText} disabled={this.props.loading} />
        </form>
      </div>
    )
  }

}

export default BeerForm
