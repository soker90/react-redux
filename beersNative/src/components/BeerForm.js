import React, { Component } from 'react'
import { Text, TextInput, TouchableOpacity, View } from 'react-native'


class BeerForm extends Component {
  constructor(props) {
    super(props)
    this.name = null
    this.graduation = null
    this.price = null
  }

  handleSubmit(e) {
    
    this.props.addBeer(this.name, this.graduation, this.price)
  }

  render() {
    const btnText = this.props.loading ? 'Loading...' : 'Add Beer'

    return (
      <View>
        <View action="#" >
          <View>
            <Text>Name </Text>
            <TextInput type="text" onChangeText={(text) => this.name = text} name="beerName" required="required"/>
          </View>
          <View>
            <Text>Graduation </Text>
            <TextInput type="text" onChangeText={(text) => this.graduation = text} name="beerGraduation" required="required"/>
          </View>
          <View>
            <Text>Price </Text>
            <TextInput type="text" onChangeText={(text) => this.price = text} name="beerPrice" required="required"/>
          </View>
          <TouchableOpacity onPress={(e) => this.handleSubmit(e)} disabled={this.props.loading}>
            <Text>{btnText}</Text>
          </TouchableOpacity>
        </View>
      </View>
    )
  }

}

export default BeerForm
