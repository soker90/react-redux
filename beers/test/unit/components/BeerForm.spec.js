// BeerForm.spec.js
import React from 'react'
import renderer from 'react-test-renderer'

import BeerForm from '../../../src/components/BeerForm'

describe('BeerForm component', () => {
  test('render BeerForm component without crashing', () => {
    const loading = false

    const tree = renderer.create(
      <BeerForm loading={loading}/>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('render BeerForm component with loading true', () => {
    const loading = true

    const tree = renderer.create(
      <BeerForm loading={loading}/>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

})
