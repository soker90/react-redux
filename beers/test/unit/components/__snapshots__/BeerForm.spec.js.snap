exports[`BeerForm component render BeerForm component with loading true 1`] = `
<div>
  <form
    action="#"
    onSubmit={[Function]}>
    <input
      required="required"
      type="text" />
    <input
      required="required"
      type="text" />
    <input
      disabled={true}
      type="submit"
      value="Loading..." />
  </form>
</div>
`;

exports[`BeerForm component render BeerForm component without crashing 1`] = `
<div>
  <form
    action="#"
    onSubmit={[Function]}>
    <input
      required="required"
      type="text" />
    <input
      required="required"
      type="text" />
    <input
      disabled={false}
      type="submit"
      value="Add Beer" />
  </form>
</div>
`;
